src/jp2image.cpp

```diff
@@ -538,7 +542,7 @@ static void boxes_check(size_t b,size_t m)
 
                             if (subBox.type == kJp2BoxTypeColorHeader) {
                                 long pad = 3;  // don't know why there are 3 padding bytes
-                                enforce(data.size_ >= pad, kerCorruptedMetadata);
+                                enforce(data.size_ >= pad + 4, kerCorruptedMetadata);
```

Comment:
Sorry to be annoying, but what's the origin of the ` + 4`? 
Is that to protect the `getULong()` on line 551?   
Might make it easier for future debugging to put a comment explaining that? What do you think?

Reply:
Good point. I have added a comment.